x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/local/LocalExporter.java

```diff
@@ -203,17 +202,13 @@ LocalBulk resolveBulk(ClusterState clusterState, boolean clusterStateChange) {
             return null;
         }
 
-        // List of templates
-        final Map<String, String> templates = Arrays.stream(MonitoringTemplateUtils.TEMPLATE_IDS)
```

Comment:
Non master doesn't need to load this in the first place, master only needs to load it once. The current code loads it on every CS update until monitoring templates are fully initialized. In x-pack ITs that spin up 6 nodes this can lead to slowness.